<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/dbellicoso/catkin_ws/src/starleth_common/starleth_description/urdf/starleth_minimal.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!--Copyright (C) 2014 by Peter Fankhauser                                   -->
<!--pfankhauser@anybotics.com                                                -->
<!--                                                                         -->
<!--This program is free software; you can redistribute it and/or modify     -->
<!--it under the terms of the Lesser GNU General Public License as published -->
<!--by the Free Software Foundation; either version 3 of the License, or     -->
<!--(at your option) any later version.                                      -->
<!--                                                                         -->
<!--This program is distributed in the hope that it will be useful,          -->
<!--but WITHOUT ANY WARRANTY; without even the implied warranty of           -->
<!--MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the             -->
<!--Lesser GNU General Public License for more details.                      -->
<!--                                                                         -->
<!--You should have received a copy of the Lesser GNU General Public License -->
<!--along with this program. If not, see <http://www.gnu.org/licenses/>.     -->
<!-- Description of the StarlETH robot with minimal configuration. -->
<robot name="starleth" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!-- Import materials/colors -->
  <material name="black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="green">
    <color rgba="0.0 0.8 0.0 1.0"/>
  </material>
  <material name="grey">
    <color rgba="0.2 0.2 0.2 1.0"/>
  </material>
  <material name="light_grey">
    <color rgba="0.4 0.4 0.4 1.0"/>
  </material>
  <material name="orange">
    <color rgba="1.0 0.423529411765 0.0392156862745 1.0"/>
  </material>
  <material name="brown">
    <color rgba="0.870588235294 0.811764705882 0.764705882353 1.0"/>
  </material>
  <material name="red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="white">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <!-- Import inertia_tensors -->
  <!-- Sphere -->
  <!-- Cuboid -->
  <!-- Cylinder -->
  <!-- Import StarlETH robot parameters -->
  <!-- Math constants -->
  <!-- Scale factor for mesh -->
  <!-- Contact model -->
  <!-- Base/Main body -->
  <!-- [m] -->
  <!-- [m] -->
  <!-- [m] -->
  <!-- [kg] -->
  <!-- Tuned, does not reflect reality!!! -->
  <!-- [kg * m^2] -->
  <!-- [kg * m^2] -->
  <!-- [kg * m^2] -->
  <!-- [m] -->
  <!-- [m] -->
  <!-- [m] -->
  <!-- [m] -->
  <!-- Hip -->
  <!-- [m] -->
  <!-- [m] -->
  <!-- [kg] -->
  <!-- [kg * m^2] -->
  <!-- [kg * m^2] -->
  <!-- [kg * m^2] -->
  <!-- [m] -->
  <!-- [m] -->
  <!-- Thigh -->
  <!-- [m] -->
  <!-- [m] -->
  <!-- [kg] -->
  <!-- [kg * m^2] -->
  <!-- [kg * m^2] -->
  <!-- [kg * m^2] -->
  <!-- [m] -->
  <!-- [m] -->
  <!-- Shank -->
  <!-- [m] -->
  <!-- [m] -->
  <!-- [kg] -->
  <!-- [kg * m^2] -->
  <!-- [kg * m^2] -->
  <!-- [kg * m^2] -->
  <!-- [m] -->
  <!-- Foot -->
  <!-- [m] -->
  <!-- [m] -->
  <!-- Joint HAA -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad/s] -->
  <!-- [Nm] -->
  <!-- [Nms/rad] -->
  <!-- [Nm] -->
  <!-- Joint HFE -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad/s] -->
  <!-- [Nm] -->
  <!-- [Nms/rad] -->
  <!-- [Nm] -->
  <!-- Joint KFE -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad] -->
  <!-- [rad/s] -->
  <!-- [Nm] -->
  <!-- [Nms/rad] -->
  <!-- [Nm] -->
  <!-- Import StarlETH robot components -->
  <!-- Useful definitions -->
  <!-- Base -->
  <!-- Hip -->
  <!-- Thigh -->
  <!-- Shank -->
  <!-- Foot -->
  <!-- Import StarlETH body description -->
  <!-- Base -->
  <!-- Attach the robot to the world link if requested -->
  <!-- Base link -->
  <link name="base">
    <visual>
      <origin rpy="1.57079632679 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://starleth_description/meshes/starleth_base.dae" scale="1 1     1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.01"/>
      <geometry>
        <box size="0.7 0.45 0.2"/>
      </geometry>
    </collision>
  </link>
  <!-- Fixed joint to add dummy inertia link -->
  <joint name="base_to_base_inertia" type="fixed">
    <parent link="base"/>
    <child link="base_inertia"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <!-- Dummy inertia link, because KDL cannot have inertia on the base link -->
  <link name="base_inertia">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.0"/>
      <mass value="20.7"/>
      <inertia ixx="0.226" ixy="0.0" ixz="0.0" iyy="0.393" iyz="0.0" izz="0.553"/>
    </inertial>
  </link>
  <gazebo reference="base">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <!-- Left fore components -->
  <!-- Hip link -->
  <link name="LF_HIP">
    <visual>
      <origin rpy="1.57079632679 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://starleth_description/meshes/starleth_hip.dae" scale="1 1     1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="1.57079632679 0 0" xyz="0 0 -0.0685"/>
      <geometry>
        <cylinder length="0.25" radius="0.02"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.055"/>
      <mass value="1.8"/>
      <inertia ixx="0.009275" ixy="0.0" ixz="0.0" iyy="0.003035" iyz="0.0" izz="0.00721"/>
    </inertial>
  </link>
  <!-- Hip joint -->
  <joint name="LF_HAA" type="revolute">
    <parent link="base"/>
    <child link="LF_HIP"/>
    <origin xyz="0.2525 0.185 0"/>
    <axis xyz="1 0 0"/>
    <limit command_effort="30" current="32" gear_velocity="10" effort="30" lower="-0.35" upper="0.45" velocity="10"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="LF_HIP">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <gazebo reference="LF_HAA">
    </gazebo>
  <!-- Thigh link -->
  <link name="LF_THIGH">
    <visual>
      <origin rpy="1.57079632679 0 0.0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://starleth_description/meshes/starleth_thigh.dae" scale="1 1     1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.1"/>
      <geometry>
        <cylinder length="0.2" radius="0.04"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.148"/>
      <mass value="0.31"/>
      <inertia ixx="0.00148976" ixy="0.0" ixz="0.0" iyy="0.00142976" iyz="0.0" izz="0.000215"/>
    </inertial>
  </link>
  <!-- Thigh joint -->
  <joint name="LF_HFE" type="revolute">
    <parent link="LF_HIP"/>
    <child link="LF_THIGH"/>
    <origin xyz="0 0 -0.0685"/>
    <axis xyz="0 1 0"/>
    <limit command_effort="30" current="32" gear_velocity="10" effort="30" lower="-0.55" upper="1.24" velocity="10"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="LF_THIGH">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <gazebo reference="LF_HFE">
    </gazebo>
  <!-- Shank link -->
  <link name="LF_SHANK">
    <visual>
      <origin rpy="1.57079632679 0.09 0.0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://starleth_description/meshes/starleth_shank.dae" scale="1 1     1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 -0.54 0" xyz="0.02 0 -0.095"/>
      <geometry>
        <cylinder length="0.12" radius="0.015"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0.44 0" xyz="0.031 0 -0.22"/>
      <geometry>
        <cylinder length="0.12" radius="0.015"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.084"/>
      <mass value="0.32"/>
      <inertia ixx="0.00112108" ixy="0.0" ixz="0.0" iyy="0.00124208" iyz="0.0" izz="0.0002"/>
    </inertial>
  </link>
  <!-- Shank joint -->
  <joint name="LF_KFE" type="revolute">
    <parent link="LF_THIGH"/>
    <child link="LF_SHANK"/>
    <origin xyz="0 0 -0.2"/>
    <axis xyz="0 1 0"/>
    <limit command_effort="30" current="32" gear_velocity="10" effort="30" lower="-2.41" upper="0.6" velocity="10"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="LF_SHANK">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <gazebo reference="LF_KFE">
    </gazebo>
  <!-- Foot link -->
  <link name="LF_FOOT">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.025"/>
      </geometry>
    </collision>
  </link>
  <!-- Foot joint -->
  <joint name="LF_ANKLE" type="fixed">
    <parent link="LF_SHANK"/>
    <child link="LF_FOOT"/>
    <origin xyz="0 0 -0.276"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="LF_FOOT">
    <!-- These parameters do not seem to affect Gazebo. -->
  </gazebo>
  <gazebo reference="LF_ANKLE">
    </gazebo>
  <!-- Right fore components -->
  <!-- Hip link -->
  <link name="RF_HIP">
    <visual>
      <origin rpy="1.57079632679 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://starleth_description/meshes/starleth_hip.dae" scale="1 1     1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="1.57079632679 0 0" xyz="0 0 -0.0685"/>
      <geometry>
        <cylinder length="0.25" radius="0.02"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.055"/>
      <mass value="1.8"/>
      <inertia ixx="0.009275" ixy="0.0" ixz="0.0" iyy="0.003035" iyz="0.0" izz="0.00721"/>
    </inertial>
  </link>
  <!-- Hip joint -->
  <joint name="RF_HAA" type="revolute">
    <parent link="base"/>
    <child link="RF_HIP"/>
    <origin xyz="0.2525 -0.185 0"/>
    <axis xyz="1 0 0"/>
    <limit command_effort="30" current="32" gear_velocity="10" effort="30" lower="-0.45" upper="0.35" velocity="10"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="RF_HIP">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <gazebo reference="RF_HAA">
    </gazebo>
  <!-- Thigh link -->
  <link name="RF_THIGH">
    <visual>
      <origin rpy="1.57079632679 0 0.0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://starleth_description/meshes/starleth_thigh.dae" scale="1 1     1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.1"/>
      <geometry>
        <cylinder length="0.2" radius="0.04"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.148"/>
      <mass value="0.31"/>
      <inertia ixx="0.00148976" ixy="0.0" ixz="0.0" iyy="0.00142976" iyz="0.0" izz="0.000215"/>
    </inertial>
  </link>
  <!-- Thigh joint -->
  <joint name="RF_HFE" type="revolute">
    <parent link="RF_HIP"/>
    <child link="RF_THIGH"/>
    <origin xyz="0 0 -0.0685"/>
    <axis xyz="0 1 0"/>
    <limit command_effort="30" current="32" gear_velocity="10" effort="30" lower="-0.55" upper="1.24" velocity="10"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="RF_THIGH">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <gazebo reference="RF_HFE">
    </gazebo>
  <!-- Shank link -->
  <link name="RF_SHANK">
    <visual>
      <origin rpy="1.57079632679 0.09 0.0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://starleth_description/meshes/starleth_shank.dae" scale="1 1     1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 -0.54 0" xyz="0.02 0 -0.095"/>
      <geometry>
        <cylinder length="0.12" radius="0.015"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0.44 0" xyz="0.031 0 -0.22"/>
      <geometry>
        <cylinder length="0.12" radius="0.015"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.084"/>
      <mass value="0.32"/>
      <inertia ixx="0.00112108" ixy="0.0" ixz="0.0" iyy="0.00124208" iyz="0.0" izz="0.0002"/>
    </inertial>
  </link>
  <!-- Shank joint -->
  <joint name="RF_KFE" type="revolute">
    <parent link="RF_THIGH"/>
    <child link="RF_SHANK"/>
    <origin xyz="0 0 -0.2"/>
    <axis xyz="0 1 0"/>
    <limit command_effort="30" current="32" gear_velocity="10" effort="30" lower="-2.41" upper="0.6" velocity="10"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="RF_SHANK">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <gazebo reference="RF_KFE">
    </gazebo>
  <!-- Foot link -->
  <link name="RF_FOOT">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.025"/>
      </geometry>
    </collision>
  </link>
  <!-- Foot joint -->
  <joint name="RF_ANKLE" type="fixed">
    <parent link="RF_SHANK"/>
    <child link="RF_FOOT"/>
    <origin xyz="0 0 -0.276"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="RF_FOOT">
    <!-- These parameters do not seem to affect Gazebo. -->
  </gazebo>
  <gazebo reference="RF_ANKLE">
    </gazebo>
  <!-- Left hind components -->
  <!-- Hip link -->
  <link name="LH_HIP">
    <visual>
      <origin rpy="1.57079632679 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://starleth_description/meshes/starleth_hip.dae" scale="1 1     1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="1.57079632679 0 0" xyz="0 0 -0.0685"/>
      <geometry>
        <cylinder length="0.25" radius="0.02"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.055"/>
      <mass value="1.8"/>
      <inertia ixx="0.009275" ixy="0.0" ixz="0.0" iyy="0.003035" iyz="0.0" izz="0.00721"/>
    </inertial>
  </link>
  <!-- Hip joint -->
  <joint name="LH_HAA" type="revolute">
    <parent link="base"/>
    <child link="LH_HIP"/>
    <origin xyz="-0.2525 0.185 0"/>
    <axis xyz="1 0 0"/>
    <limit command_effort="30" current="32" gear_velocity="10" effort="30" lower="-0.35" upper="0.45" velocity="10"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="LH_HIP">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <gazebo reference="LH_HAA">
    </gazebo>
  <!-- Thigh link -->
  <link name="LH_THIGH">
    <visual>
      <origin rpy="1.57079632679 0 3.14159265359" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://starleth_description/meshes/starleth_thigh.dae" scale="1 1     1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.1"/>
      <geometry>
        <cylinder length="0.2" radius="0.04"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.148"/>
      <mass value="0.31"/>
      <inertia ixx="0.00148976" ixy="0.0" ixz="0.0" iyy="0.00142976" iyz="0.0" izz="0.000215"/>
    </inertial>
  </link>
  <!-- Thigh joint -->
  <joint name="LH_HFE" type="revolute">
    <parent link="LH_HIP"/>
    <child link="LH_THIGH"/>
    <origin xyz="0 0 -0.0685"/>
    <axis xyz="0 1 0"/>
    <limit command_effort="30" current="32" gear_velocity="10" effort="30" lower="-1.24" upper="0.55" velocity="10"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="LH_THIGH">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <gazebo reference="LH_HFE">
    </gazebo>
  <!-- Shank link -->
  <link name="LH_SHANK">
    <visual>
      <origin rpy="1.57079632679 0.09 3.14159265359" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://starleth_description/meshes/starleth_shank.dae" scale="1 1     1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0.54 0" xyz="-0.02 0 -0.095"/>
      <geometry>
        <cylinder length="0.12" radius="0.015"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 -0.44 0" xyz="-0.031 0 -0.22"/>
      <geometry>
        <cylinder length="0.12" radius="0.015"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.084"/>
      <mass value="0.32"/>
      <inertia ixx="0.00112108" ixy="0.0" ixz="0.0" iyy="0.00124208" iyz="0.0" izz="0.0002"/>
    </inertial>
  </link>
  <!-- Shank joint -->
  <joint name="LH_KFE" type="revolute">
    <parent link="LH_THIGH"/>
    <child link="LH_SHANK"/>
    <origin xyz="0 0 -0.2"/>
    <axis xyz="0 1 0"/>
    <limit command_effort="30" current="32" gear_velocity="10" effort="30" lower="-0.6" upper="2.41" velocity="10"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="LH_SHANK">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <gazebo reference="LH_KFE">
    </gazebo>
  <!-- Foot link -->
  <link name="LH_FOOT">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.025"/>
      </geometry>
    </collision>
  </link>
  <!-- Foot joint -->
  <joint name="LH_ANKLE" type="fixed">
    <parent link="LH_SHANK"/>
    <child link="LH_FOOT"/>
    <origin xyz="0 0 -0.276"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="LH_FOOT">
    <!-- These parameters do not seem to affect Gazebo. -->
  </gazebo>
  <gazebo reference="LH_ANKLE">
    </gazebo>
  <!-- Right hind components -->
  <!-- Hip link -->
  <link name="RH_HIP">
    <visual>
      <origin rpy="1.57079632679 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://starleth_description/meshes/starleth_hip.dae" scale="1 1     1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="1.57079632679 0 0" xyz="0 0 -0.0685"/>
      <geometry>
        <cylinder length="0.25" radius="0.02"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.055"/>
      <mass value="1.8"/>
      <inertia ixx="0.009275" ixy="0.0" ixz="0.0" iyy="0.003035" iyz="0.0" izz="0.00721"/>
    </inertial>
  </link>
  <!-- Hip joint -->
  <joint name="RH_HAA" type="revolute">
    <parent link="base"/>
    <child link="RH_HIP"/>
    <origin xyz="-0.2525 -0.185 0"/>
    <axis xyz="1 0 0"/>
    <limit command_effort="30" current="32" gear_velocity="10" effort="30" lower="-0.45" upper="0.35" velocity="10"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="RH_HIP">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <gazebo reference="RH_HAA">
    </gazebo>
  <!-- Thigh link -->
  <link name="RH_THIGH">
    <visual>
      <origin rpy="1.57079632679 0 3.14159265359" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://starleth_description/meshes/starleth_thigh.dae" scale="1 1     1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.1"/>
      <geometry>
        <cylinder length="0.2" radius="0.04"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.148"/>
      <mass value="0.31"/>
      <inertia ixx="0.00148976" ixy="0.0" ixz="0.0" iyy="0.00142976" iyz="0.0" izz="0.000215"/>
    </inertial>
  </link>
  <!-- Thigh joint -->
  <joint name="RH_HFE" type="revolute">
    <parent link="RH_HIP"/>
    <child link="RH_THIGH"/>
    <origin xyz="0 0 -0.0685"/>
    <axis xyz="0 1 0"/>
    <limit command_effort="30" current="32" gear_velocity="10" effort="30" lower="-1.24" upper="0.55" velocity="10"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="RH_THIGH">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <gazebo reference="RH_HFE">
    </gazebo>
  <!-- Shank link -->
  <link name="RH_SHANK">
    <visual>
      <origin rpy="1.57079632679 0.09 3.14159265359" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://starleth_description/meshes/starleth_shank.dae" scale="1 1     1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0.54 0" xyz="-0.02 0 -0.095"/>
      <geometry>
        <cylinder length="0.12" radius="0.015"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 -0.44 0" xyz="-0.031 0 -0.22"/>
      <geometry>
        <cylinder length="0.12" radius="0.015"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.084"/>
      <mass value="0.32"/>
      <inertia ixx="0.00112108" ixy="0.0" ixz="0.0" iyy="0.00124208" iyz="0.0" izz="0.0002"/>
    </inertial>
  </link>
  <!-- Shank joint -->
  <joint name="RH_KFE" type="revolute">
    <parent link="RH_THIGH"/>
    <child link="RH_SHANK"/>
    <origin xyz="0 0 -0.2"/>
    <axis xyz="0 1 0"/>
    <limit command_effort="30" current="32" gear_velocity="10" effort="30" lower="-0.6" upper="2.41" velocity="10"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="RH_SHANK">
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <gazebo reference="RH_KFE">
    </gazebo>
  <!-- Foot link -->
  <link name="RH_FOOT">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.025"/>
      </geometry>
    </collision>
  </link>
  <!-- Foot joint -->
  <joint name="RH_ANKLE" type="fixed">
    <parent link="RH_SHANK"/>
    <child link="RH_FOOT"/>
    <origin xyz="0 0 -0.276"/>
  </joint>
  <!-- Gazebo customization -->
  <gazebo reference="RH_FOOT">
    <!-- These parameters do not seem to affect Gazebo. -->
  </gazebo>
  <gazebo reference="RH_ANKLE">
    </gazebo>
  <!-- Import StarlETH Gazebo controller -->
  <gazebo>
    <plugin filename="libstarleth_gazebo_ros_control.so" name="gazebo_ros_control">
      <robotNamespace>starleth</robotNamespace>
      <!-- Currently unused -->
      <robotDescription>anymal_description</robotDescription>
      <robotBaseLink>base</robotBaseLink>
      <statePublisherRate>400.0</statePublisherRate>
    </plugin>
  </gazebo>
  <!-- Xsens Mti
  <xacro:include filename="$(find any_description)/urdf/sensors/imu/xsens_mti/xsens_mti.urdf.xacro"/>
  <xacro:xsens_mti name="imu" parent="base" update_rate="400" ns="sensors" frame_id="/imu_link" topic_name="imu">
    <origin xyz="${l_b * 0.5} 0 ${collision_h_b + collision_z_offset_b - 0.01}"/>
  </xacro:xsens_mti>
-->
</robot>
